### PR TITLE
MANTA-4965 manta-deploy-lab should support multiple servers

### PR DESCRIPTION
**Pull request by trentm**
This updates 'manta-deploy-dev' to take and pass through args to
'manta-adm genconfig ... lab' for laying out a Manta "lab" size
deployment across multiple servers. The new lab-size options are:

    -s SERVERS, --servers=SERVERS
                       Specify servers to use for Manta service instances.
                       SERVERS a comma-separated list of server UUIDs. If not
                       specified, the deployment will use *all* available
                       servers.
    -S SERVERS, --storage-servers=SERVERS
                       Specify servers to use for Manta "storage" service
                       instances. SERVERS a comma-separated list of server
                       UUIDs. If not specified, the servers given by "--servers"
                       will be used.
    -1 SVCNAMES, --one-on-headnode=SVCNAMES
                       Specify service names to ensure at least one instance is
                       deployed to the headnode. This is for developer
                       convenience. If not specified, it defaults to:
                       "loadbalancer,webapi,buckets-api"

This also notably changes the default "lab" size to use *all available
setup servers*.

Also, this changes 'manta-adm show -js' and 'manta-adm genconfig lab'
output to be *sorted* (by CN UUID and by service name) so they can
more meaningfully be compared.

### CODE REVIEW

File: lib/adm.js [hunk 1]
```diff
@@ -2385,36 +2394,306 @@ maAdm.prototype.dumpConfigLab = function(options, callback) {
...
+
+                // Which server for instances? Some guidelines:
+                // - Spread "storage" insts across all "storageServers" and warn if
+                //   there aren't enough servers.
+                // - Spread other insts of a given service across all "servers" and
+                //   warn if there aren't enough servers.
+                // - Bias (for now) to having one of everything on the headnode to
+                //   facilitate dev/debugging, if the headnode is one of the
+                //   servers.
+                function createLayout(ctx, next) {
```
[the remark this answers is not in the document]
Comment (trentm) [reply lib/adm.js]: > I think I would expect that if the number of storage servers is less than CNs + HN, that the HN _does not_ get a storage instance

That would be a nice thing to do, but this change doesn't tackle that at all.


> I guess this is just generating the config though so the operator can always fix that themselves.

Sorta. If used as part of `manta-deploy-lab` it will go ahead and create that storage zone on the headnode, potentially.

Anyway, I'm calling this one out of scope.



> Overall I find this function kinda long and hard to follow and probably could at least use a comment explaining its algorithms.

Thanks, yes I'll try to document it.

**Comment by trentm** [comment]
@joshwilsdon  Do these comments help, or still confusing?

**Comment by joshwilsdon** [comment]
> @joshwilsdon Do these comments help, or still confusing?

It looks like part of your change didn't get committed? The comment ends with:

```
How it works: It
```

**Comment by trentm** [comment]
> It looks like part of your change didn't get committed?

Oops. Thanks. You caught me rushing there. I'd started describing at the top of the function, but moved the description down next to the for-loop.
